### PR TITLE
fix: revert to legacy concatenate plus small fixes.

### PR DESCRIPTION
**Pull request by FedeDP**

### CODE REVIEW

File: userspace/libsinsp/test/sinsp_utils.ut.cpp
Comment (FedeDP) [inline userspace/libsinsp/test/sinsp_utils.ut.cpp:1]: Added and adapted tests.

File: userspace/libsinsp/unix_paths.cpp [hunk 1]
```diff
@@ -45,7 +45,7 @@ void rewind_to_parent_path(char* targetbase, char** tc, const char** pc, uint32_
 
 	(*tc)--;
 
-	while(*((*tc) - 1) != '/' && (*tc) >= targetbase + 1)
+	while((*tc) >= targetbase + 1 && *((*tc) - 1) != '/')
```
Comment (FedeDP) [inline userspace/libsinsp/unix_paths.cpp:48]: Inverted check to avoid possible underflow.

File: userspace/libsinsp/unix_paths.cpp [hunk 2]
```diff
@@ -142,9 +144,13 @@ void copy_and_sanitize_path(char* target, char* targetbase, const char* path, ch
...
 				//
-				// separator, if the last char is already a separator, skip it
+				// separator:
+				// * if the last char is already a separator, skip it
+				// * if we are back at targetbase but targetbase was not empty before, it means we
+				//   fully rewinded back to targetbase and the string is now empty. Skip separator.
+				//   Example: "/foo/../a" -> "/a" BUT "foo/../a" -> "a"
 				//
-				if(tc > targetbase && *(tc - 1) == separator)
+				if((tc > targetbase && *(tc - 1) == separator) || (tc == targetbase && !empty_base))
```
Comment (FedeDP) [inline userspace/libsinsp/unix_paths.cpp:154]: When cutting away `../` in the target string, we return to its start, avoid putting a `/`; this fixes edge cases like:
```
"foo/../a"
```
that previously returned `/a` instead of just `a`.

Comment (FedeDP) [reply userspace/libsinsp/unix_paths.cpp:154]: This behavior is covered by tests.
